Userland/Libraries/LibCore/MemoryStream.h

```diff
@@ -29,15 +29,15 @@ class MemoryStream final : public SeekableStream {
     // FIXME: It doesn't make sense to truncate a memory stream. Therefore, we don't do anything here. Is that fine?
     virtual ErrorOr<void> truncate(off_t) override { return Error::from_errno(ENOTSUP); }
 
-    virtual ErrorOr<size_t> read(Bytes bytes) override
+    virtual ErrorOr<Bytes> read(Bytes bytes) override
     {
         auto to_read = min(remaining(), bytes.size());
         if (to_read == 0)
-            return 0;
+            return Bytes {};
 
         m_bytes.slice(m_offset, to_read).copy_to(bytes);
         m_offset += to_read;
-        return bytes.size();
+        return Bytes { bytes.data(), to_read };
```

Comment:
Super minor, but it might be a bit clearer that we're returning a slice of the input buffer if we do (here and in other `read` methods):
```c++
return bytes.trim(to_read);
```

Reply:
Thanks, I didn't notice we could do that, it's definitely better. :^)